src/util/workers.c

```diff
@@ -30,33 +31,58 @@ static void yieldCallback(void *yieldCtx) {
   RedisModule_Yield(ctx, REDISMODULE_YIELD_FLAG_CLIENTS, NULL);
 }
 
+/* Configure here anything that needs to know it can use the thread pool */
+static void workersThreadPool_OnTurnOn(size_t new_num) {
+  // Log that we've enabled the thread pool.
+  RedisModule_Log(RSDummyContext, "notice", "Enabled workers threadpool of size %lu", new_num);
+  // Change VecSim write mode temporarily for fast RDB loading of vector index (if needed).
+  VecSim_SetWriteMode(VecSim_WriteAsync);
+}
+
+/* Configure here anything that needs to know it cannot use the thread pool anymore */
+static void workersThreadPool_OnTurnOff(size_t old_num) {
+  RedisModule_Log(RSDummyContext, "notice", "Disabled workers threadpool of size %lu", old_num);
+  VecSim_SetWriteMode(VecSim_WriteInPlace);
+}
+
 // set up workers' thread pool
 int workersThreadPool_CreatePool(size_t worker_count) {
   assert(_workers_thpool == NULL);
 
   _workers_thpool = redisearch_thpool_create(worker_count, RSGlobalConfig.highPriorityBiasNum, LogCallback, "workers");
   if (_workers_thpool == NULL) return REDISMODULE_ERR;
-
+  if (worker_count > 0) {
+    workersThreadPool_OnTurnOn(worker_count);
+  } else {
+    workersThreadPool_OnTurnOff(worker_count);
+  }
   return REDISMODULE_OK;
 }
 
-size_t workersThreadPool_SetNumWorkers(size_t worker_count) {
-  if(_workers_thpool == NULL) return worker_count;
+void workersThreadPool_SetNumWorkers() {
```

Comment:
IMO sending worker_count as a paramter satisfies a more readable and robust API
Why not to move setting the number of threads in event case to workersThreadPool_OnEventStart?
I think it will make this function more clean.

Reply:
I want this function to be the single point for all the number of threads logic. It is called from both new configurations and when an event starts or ends. passing the number of workers requires deciding what the new value should be, and we don't want to have this logic in 4 different places before calling the function.
I can add a description of the global parameters this function uses to make its decision.